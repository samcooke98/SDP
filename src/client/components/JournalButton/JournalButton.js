import React from "react";
import styles from "./web.css";
import moment from 'moment';

import Menu from "react-icons/lib/fa/ellipsis-v";

import Text from "../Text/Text.js";
import SVG from "./MyComponent.js";

export default class JournalButton extends React.Component {


    render() {
        console.log(this.props);
        return (
            <div className={styles.journalWrap}>
                <div style={{cursor: "pointer"}} onClick={this.props.onPress} className={styles.journalbtn}>
                    {/* <object data={require("./book.svg")} style={{fill: 'red'}}/>  */}
                    <SVG height="140" width="100" colour={this.props.colour} />
                    <div style={{ display: 'flex', width: '100%', justifyContent: "center", alignItems: "center"}}>
                        <span style={{ textAlign: 'center', flexGrow: 1 }}>
                            <Text tag='h3'> {this.props.title} </Text>
                            <Text>Created: {moment(this.props.date).format("DD/MM/YYYY")} </Text>
                        </span>
                        <span style={{ width: 0, overflow: 'visible' }}>
                            <Menu height={32} width={32} onClick={(evt) => { console.log("OPEN MENU TODO"); evt.stopPropagation() }}
                                style={{ position: 'relative', right: "32px" }}
                            />
                        </span>
                    </div>
                </div>
            </div>

        )

    }

    someFunc = () => {
        //'This' is automatically bound when writing functions like this
        return JSON.stringify(this.props);
    }
}


//In addition to CSS modules, you may wish to use: https://github.com/gajus/react-css-modules

