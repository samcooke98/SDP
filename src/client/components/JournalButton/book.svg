<svg width="129" height="169" viewBox="0 0 129 169" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
<title>Group</title>
<desc>Created using Figma</desc>
<g id="Canvas" transform="translate(-1563 1483)">
<g id="Group">
<g id="Rectangle 7">
<use xlink:href="#path0_fill" transform="translate(1572.28 -1483)" fill="#333333"/>
</g>
<g id="Rectangle 8">
<use xlink:href="#path1_fill" transform="translate(1563 -1455.35)" fill="#F8F8F8"/>
<mask id="mask0_outline_ins">
<use xlink:href="#path1_fill" fill="white" transform="translate(1563 -1455.35)"/>
</mask>
<g mask="url(#mask0_outline_ins)">
<use xlink:href="#path2_stroke_2x" transform="translate(1563 -1455.35)" fill="#333333"/>
</g>
</g>
<g id="Rectangle 8">
<use xlink:href="#path1_fill" transform="translate(1563 -1405.58)" fill="#F8F8F8"/>
<mask id="mask1_outline_ins">
<use xlink:href="#path1_fill" fill="white" transform="translate(1563 -1405.58)"/>
</mask>
<g mask="url(#mask1_outline_ins)">
<use xlink:href="#path2_stroke_2x" transform="translate(1563 -1405.58)" fill="#333333"/>
</g>
</g>
<g id="Rectangle 8">
<use xlink:href="#path1_fill" transform="translate(1563 -1355.97)" fill="#F8F8F8"/>
<mask id="mask2_outline_ins">
<use xlink:href="#path1_fill" fill="white" transform="translate(1563 -1355.97)"/>
</mask>
<g mask="url(#mask2_outline_ins)">
<use xlink:href="#path2_stroke_2x" transform="translate(1563 -1355.97)" fill="#333333"/>
</g>
</g>
<g id="Rectangle 9">
<use xlink:href="#path3_fill" transform="translate(1599 -1451.28)" fill="#F8F8F8"/>
</g>
</g>
</g>
<defs>
<path id="path0_fill" d="M 0 0L 116.603 0C 118.322 0 119.716 1.39391 119.716 3.11338L 119.716 169L 0 169L 0 0Z"/>
<path id="path1_fill" d="M 0 0L 18.5682 0L 18.5682 14.3502L 0 14.3502L 0 0Z"/>
<path id="path2_stroke_2x" d="M 0 0L 0 -4.06919L -4.06919 -4.06919L -4.06919 0L 0 0ZM 18.5682 0L 22.6374 0L 22.6374 -4.06919L 18.5682 -4.06919L 18.5682 0ZM 18.5682 14.3502L 18.5682 18.4194L 22.6374 18.4194L 22.6374 14.3502L 18.5682 14.3502ZM 0 14.3502L -4.06919 14.3502L -4.06919 18.4194L 0 18.4194L 0 14.3502ZM 0 4.06919L 18.5682 4.06919L 18.5682 -4.06919L 0 -4.06919L 0 4.06919ZM 14.499 0L 14.499 14.3502L 22.6374 14.3502L 22.6374 0L 14.499 0ZM 18.5682 10.281L 0 10.281L 0 18.4194L 18.5682 18.4194L 18.5682 10.281ZM 4.06919 14.3502L 4.06919 0L -4.06919 0L -4.06919 14.3502L 4.06919 14.3502Z"/>
<path id="path3_fill" d="M 0 0L 66.2917 0L 66.2917 26.025L 0 26.025L 0 0Z"/>
</defs>
</svg>
