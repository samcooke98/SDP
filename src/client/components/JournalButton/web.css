.journalbtn { 
    margin-left:auto;
    margin-right:auto;

    display:flex;
    flex-direction:column;
    align-items: center;
    justify-content: center;
    padding: 6px;
    background-color: rgb(230,233,237);


    width: 300px;
    height: 244px; 

    box-shadow: 0px 4px 4px rgba(0,0,0,0.25);
    border-radius: 5px;
    cursor: pointer;
}

.journalWrap { 
    width: 33%;
    margin-top: 12px;
    margin-bottom: 12px;    
}

.journalbtn img { 
    margin-top: 24px;
    height: 140px;
    width: auto;    
}

.journalTitle{ 
    font-family: "Ralway";
    font-weight: bold;
    font-size: 36px;
    text-align:center;

    margin:0;
    margin-top: 8px;
}

.caption { 
        text-align:center;
    margin: 0;
    font-family: "Raleway";
    font-size: 20px;
}