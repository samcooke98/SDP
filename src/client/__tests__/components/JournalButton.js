import React from "react";
import renderer from "react-test-renderer";
// import ReactDOM from 'react-dom';

import JournalButton from "../../components/JournalButton/JournalButton";


describe('React Component: JournalButton', () => {
    it("matches a snapshot", () => {
        const tree = renderer.create(
            <JournalButton />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})