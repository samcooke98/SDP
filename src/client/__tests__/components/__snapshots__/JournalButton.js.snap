// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Component: JournalButton matches a snapshot 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
    onClick={undefined}
    style={
      Object {
        "cursor": "pointer",
      }
    }
  >
    <svg
      height="140"
      version="1.1"
      viewBox="0 0 129 169"
      width="100"
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <title>
        Group
      </title>
      <desc>
        Created using Figma
      </desc>
      <g
        id="Canvas"
        transform="translate(-1563 1483)"
      >
        <g
          id="Group"
        >
          <g
            id="Rectangle 7"
          >
            <use
              fill={undefined}
              transform="translate(1572.28 -1483)"
              xlinkHref="#path0_fill"
            />
          </g>
          <g
            id="Rectangle 8"
          >
            <use
              fill="#F8F8F8"
              transform="translate(1563 -1455.35)"
              xlinkHref="#path1_fill"
            />
            <mask
              id="mask0_outline_ins"
            >
              <use
                fill="white"
                transform="translate(1563 -1455.35)"
                xlinkHref="#path1_fill"
              />
            </mask>
            <g
              mask="url(#mask0_outline_ins)"
            >
              <use
                fill={undefined}
                transform="translate(1563 -1455.35)"
                xlinkHref="#path2_stroke_2x"
              />
            </g>
          </g>
          <g
            id="Rectangle 8"
          >
            <use
              fill="#F8F8F8"
              transform="translate(1563 -1405.58)"
              xlinkHref="#path1_fill"
            />
            <mask
              id="mask1_outline_ins"
            >
              <use
                fill="white"
                transform="translate(1563 -1405.58)"
                xlinkHref="#path1_fill"
              />
            </mask>
            <g
              mask="url(#mask1_outline_ins)"
            >
              <use
                fill={undefined}
                transform="translate(1563 -1405.58)"
                xlinkHref="#path2_stroke_2x"
              />
            </g>
          </g>
          <g
            id="Rectangle 8"
          >
            <use
              fill="#F8F8F8"
              transform="translate(1563 -1355.97)"
              xlinkHref="#path1_fill"
            />
            <mask
              id="mask2_outline_ins"
            >
              <use
                fill="white"
                transform="translate(1563 -1355.97)"
                xlinkHref="#path1_fill"
              />
            </mask>
            <g
              mask="url(#mask2_outline_ins)"
            >
              <use
                fill={undefined}
                transform="translate(1563 -1355.97)"
                xlinkHref="#path2_stroke_2x"
              />
            </g>
          </g>
          <g
            id="Rectangle 9"
          >
            <use
              fill="#F8F8F8"
              transform="translate(1599 -1451.28)"
              xlinkHref="#path3_fill"
            />
          </g>
        </g>
      </g>
      <defs>
        <path
          d="M 0 0L 116.603 0C 118.322 0 119.716 1.39391 119.716 3.11338L 119.716 169L 0 169L 0 0Z"
          id="path0_fill"
        />
        <path
          d="M 0 0L 18.5682 0L 18.5682 14.3502L 0 14.3502L 0 0Z"
          id="path1_fill"
        />
        <path
          d="M 0 0L 0 -4.06919L -4.06919 -4.06919L -4.06919 0L 0 0ZM 18.5682 0L 22.6374 0L 22.6374 -4.06919L 18.5682 -4.06919L 18.5682 0ZM 18.5682 14.3502L 18.5682 18.4194L 22.6374 18.4194L 22.6374 14.3502L 18.5682 14.3502ZM 0 14.3502L -4.06919 14.3502L -4.06919 18.4194L 0 18.4194L 0 14.3502ZM 0 4.06919L 18.5682 4.06919L 18.5682 -4.06919L 0 -4.06919L 0 4.06919ZM 14.499 0L 14.499 14.3502L 22.6374 14.3502L 22.6374 0L 14.499 0ZM 18.5682 10.281L 0 10.281L 0 18.4194L 18.5682 18.4194L 18.5682 10.281ZM 4.06919 14.3502L 4.06919 0L -4.06919 0L -4.06919 14.3502L 4.06919 14.3502Z"
          id="path2_stroke_2x"
        />
        <path
          d="M 0 0L 66.2917 0L 66.2917 26.025L 0 26.025L 0 0Z"
          id="path3_fill"
        />
      </defs>
    </svg>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <span
        style={
          Object {
            "flexGrow": 1,
            "textAlign": "center",
          }
        }
      >
        <h3
          style={
            Object {
              "fontFamily": "Raleway, sans-serif",
            }
          }
        >
           
           
           
           
        </h3>
        <p
          style={
            Object {
              "fontFamily": "Raleway, sans-serif",
            }
          }
        >
           
          Created: 
          09/10/2017
           
           
        </p>
      </span>
      <span
        style={
          Object {
            "overflow": "visible",
            "width": 0,
          }
        }
      >
        <svg
          fill="currentColor"
          height={32}
          onClick={[Function]}
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "position": "relative",
              "right": "32px",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width={32}
        >
          <g>
            <path
              d="m24.1 27.9v4.2q0 0.9-0.7 1.6t-1.5 0.6h-4.3q-0.9 0-1.5-0.6t-0.6-1.6v-4.2q0-0.9 0.6-1.6t1.5-0.6h4.3q0.9 0 1.5 0.6t0.7 1.6z m0-11.5v4.3q0 0.9-0.7 1.5t-1.5 0.7h-4.3q-0.9 0-1.5-0.7t-0.6-1.5v-4.3q0-0.9 0.6-1.5t1.5-0.6h4.3q0.9 0 1.5 0.6t0.7 1.5z m0-11.4v4.3q0 0.9-0.7 1.5t-1.5 0.6h-4.3q-0.9 0-1.5-0.6t-0.6-1.5v-4.3q0-0.9 0.6-1.5t1.5-0.6h4.3q0.9 0 1.5 0.6t0.7 1.5z"
            />
          </g>
        </svg>
      </span>
    </div>
  </div>
</div>
`;
